### ISSUE TITLE
Synchronous dispatch of MULTI returns null

### ISSUE THREAD

**Issue by tzxyz** [lettuce-io/lettuce-core#1335]
#### Current Behavior

``` java
public class LettuceTest {

    private final RedisClient client = RedisClient.create("redis://localhost");

    private final StatefulRedisConnection<String, String> connection = client.connect();

    @Test
    public void test() {
        String result = connection.sync().dispatch(CommandType.MULTI, new StatusOutput<String, String>(StringCodec.ASCII));
        Assert.assertSame("OK", result);	// expected return "OK" but null
    }
}
```

<details>
<summary>Stack trace</summary>

```java
// your stack trace here;
```

</details>

#### Input Code

<details>
<summary>Input Code</summary>

```  java
        String result = connection.sync().multi();    // return "OK" 
        String customCommandResult = connection.sync().dispatch(CommandType.MULTI, new StatusOutput<String, String>(StringCodec.ASCII));    // return null
```

</details>

#### Environment

- Lettuce version(s): [e.g. 5.2.1.RELEASE]
- Redis version: [e.g. 4.0.10]

#### Possible Solution

in io.lettuce.core.FutureSyncInvocationHandler#isTransactionActive it shoule be
``` java
    private static boolean isNonTxControlMethod(String methodName, Object[] args) {
        return !methodName.equals("exec")
                && !methodName.equals("multi")
                && !methodName.equals("discard")
                && !(methodName.equals("dispatch") && args.length > 0 && args[0] == CommandType.MULTI)
                && !(methodName.equals("dispatch") && args.length > 0 && args[0] == CommandType.EXEC)
                && !(methodName.equals("dispatch") && args.length > 0 && args[0] == CommandType.DISCARD);
    }
```

**Comment by mp911de** [comment]
Thanks for raising the issue. It makes sense to return responses as we do the same for the regular API.

**Comment by mp911de** [comment]
Fixed via #1336.
